### PR TITLE
Get latest secret

### PR DESCRIPTION
**Pull request by rosasck**
Gets the latest version of a secret or empty if the secret doesn't exist
tests included

### CODE REVIEW

File: __tests__/secret.test.js
Comment (galxy25) [inline __tests__/secret.test.js:204]: `gets`

File: __tests__/secret.test.js
Comment (galxy25) [inline __tests__/secret.test.js:180]: `gets`

File: lib/identity/client.js
Comment (galxy25) [inline lib/identity/client.js:176]: so I guess the caller of this code just always has to check this response is not null? Can we return a struct that has a field for if the secret exists at all, and a field for holding that secret (if it does exist)

File: lib/identity/client.js
Comment (galxy25) [inline lib/identity/client.js:170]: Since search is sorting these ascending, wouldn't the first secret it returned be the oldest version (not the newest)?

Comment (galxy25) [reply lib/identity/client.js:170]: https://github.com/tozny/e3dbSearchService/blob/master/clients/elastic_query.go#L134
https://github.com/tozny/api-specs/blob/master/search/swagger.yaml#L241

Comment (rosasck) [reply lib/identity/client.js:170]: Yeah! my work around was to take the last record returned

Comment (galxy25) [reply lib/identity/client.js:170]: Prefer updating search as discussed https://tozny.slack.com/archives/G0DPBU8T1/p1612902038062200 to support this use case (we will definitely re-use & benefit from that search capability in other areas)

You're current code is making the assumption that there isn't more than a few hundred versions of the secret, instead of checking to see if it needs to page to get more search results, but if you update search you won't have to worry about that ;-)

File: lib/identity/client.js
Comment (galxy25) [inline lib/identity/client.js:176]: can result be null instead / won't this blow up if the list is empty and you try and index the first item (which won't exist because its empty...)

File: lib/identity/client.js
Comment (rosasck) [reply lib/identity/client.js:170]: i read that message wrong the first time, no wonder trying to get descending to work, did not

File: types/searchRange.js
Comment (efabens) [inline types/searchRange.js:10]: The new parameter should be placed at the end so that existing implementations don't get unexpected behavior

File: lib/identity/client.js
Comment (efabens) [inline lib/identity/client.js:180]: Should this now have a range that is descending?

File: lib/identity/client.js
Comment (efabens) [inline lib/identity/client.js:185]: I think we want to be able to grab item 0 here such that we know even if there are multiple "pages" of results we have the newest

File: lib/identity/client.js
Comment (efabens) [inline lib/identity/client.js:183]: errant console log that should get removed before merging

**Comment by rosasck** [comment]
Reworked for changes with Luke and now it works and uses the search end point changes

File: types/searchRange.js
Comment (efabens) [inline types/searchRange.js:20]: Doesn't this still need the new constructor parameter?

File: __tests__/secret.test.js
Comment (galxy25) [inline __tests__/secret.test.js:234]: should this be `it doesn't`?

File: lib/identity/client.js
Comment (galxy25) [inline lib/identity/client.js:169]: `getLatestSecret`

**Review by galxy25 — APPROVED** [review]

File: types/searchRange.js
Comment (rosasck) [reply types/searchRange.js:20]: We took it out of SearchRange and put it in SearchOrder

File: __tests__/secret.test.js
Comment (lkwdwrd) [inline __tests__/secret.test.js:214]: I see now why the waitForNext method would not work here... The functional programmer in me says there is a pattern we can abstract out to be even more general, but let's hold off until we see another area that is needed :)

Good job adapting the retry loop for your needs!

File: __tests__/utils/operations.js
Comment (lkwdwrd) [inline __tests__/utils/operations.js:775]: 👍 - yay, proper types after transport!

File: __tests__/secret.test.js
Comment (galxy25) [reply __tests__/secret.test.js:214]: Lol I had to resist the urge to @ you on this and ask for how to do this with currying, I know it's possible in js

File: types/search.js
Comment (lkwdwrd) [inline types/search.js:36]: I think this chunk for range can get reverted with the `searchOrder` type, yes?

**Review by lkwdwrd — APPROVED** [review]
Looking good. Some minor cleanup, but mostly kudos!

**Review by efabens — APPROVED** [review]

**Pull request merged**
